### ISSUE TITLE
The ciou is better than giou ?

### ISSUE THREAD

**Issue by brian208579** [AlexeyAB/darknet#5218]
Hi @AlexeyAB 
I follow this [paper](https://arxiv.org/pdf/1911.08287.pdf) and [implementation](https://github.com/AlexeyAB/darknet/issues/4360)
Using giou and ciou respectively for comparison
In the yolov3-tiny.cfg-file, the net setting is :
```
[net]
# Testing
# batch=1
# subdivisions=1
# Training
batch=64
subdivisions=4
width=416
height=416
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1
mosaic=1

learning_rate=0.001
burn_in=1000
max_batches = 20000
policy=steps
steps=16000,18000
scales=.1,.1
```
and the yolo layers setting is :
```
# giou
[yolo]
mask = 0,1,2
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=3
num=6
jitter=.3
random=1
truth_thresh=1
ignore_thresh=.9
iou_normalizer=0.5
iou_loss=giou
```
```
# ciou
[yolo]
mask = 0,1,2
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=3
num=6
jitter=.3
random=1
truth_thresh=1
ignore_thresh=.9
iou_normalizer=0.5
iou_loss=ciou
```
And the map result is :
### **giou**
![yolov3-tiny (2)](https://user-images.githubusercontent.com/43407888/79069870-4047a600-7d04-11ea-89e3-54a431c7263d.png)
### **ciou**
![yolov3-tiny (1)](https://user-images.githubusercontent.com/43407888/79069858-1d1cf680-7d04-11ea-8bdc-1717448358c1.png)

I think there is no difference between giou and ciou
This is slightly different from the results of the experiment on paper
Did I make any mistake ?
Could you give me some suggestions to improve the map of my result ?
Appreciate ~

**Comment by AlexeyAB** [comment]
It depends on dataset, for good test you should use large dataset.

GIoU is similar to CIoU for the final accuracy, just CIoU usually achieves the same accuracy faster and CIoU is more stable.

**Comment by brian208579** [comment]
@AlexeyAB 
Thanks your explain.
Except dataset , Do you think it depends on the model complexity ?

**Comment by AlexeyAB** [comment]
> Except dataset , Do you think it depends on the model complexity ?

I don't know.

> For example, it will be more effective when used on yolov3 ?

No.
